apps/platform-app/src/app/(files)/_actions/uploadCVdocumentFile.ts

```diff
@@ -0,0 +1,60 @@
+'use server'
+
+import { withSentry } from '@/utils/errHandling'
+import { validateCvFile } from '@/utils/validateCvFileHelper'
+import { put } from '@vercel/blob'
+import { customAlphabet } from 'nanoid'
+import { revalidatePath } from 'next/cache'
+
+const nanoid = customAlphabet(
+  '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz',
+  7,
+) // 7-character random string
+
+function mapMimeTypeToCvDocumentType(mimeType: string): 'PDF' | null {
+  if (mimeType === 'application/pdf') {
+    return 'PDF'
+  } else {
+    return null
+  }
+}
+
+export const uploadCVdocumentFile = withSentry(async (form: FormData) => {
+  const cvFile = form.get('cvFileUpload') as File | null
+
+  if (!cvFile) {
+    throw new Error('No file uploaded')
+  }
+
+  try {
+    const validatedCVfile = validateCvFile(cvFile)
+    if (!validatedCVfile) {
+      throw new Error('Invalid file')
```

Comment:
**nit:** we should return plain objects from server actions, even on errors because Next wont render error message this way because of security concerns. Eg. to prevent leaking secrets. Let's change everywhere in action to:

```ts
    if (!validatedCVfile) {
      return {
          error: "Invalid file"
       }
}
```